### ISSUE TITLE
how to assign reviewers to a PR?

### ISSUE THREAD

**Issue by dcu** [davidji99/go-bitbucket#36]

**Comment by davidji99** [comment]
Hi @dcu,

Ah I forgot to add that to `pull_requests.go`. I would need to update `UpdatePullRequestOpts` to include a reviewers field so one can add reviewers as part of the pull request `UPDATE` call. I can probably get to this in the next day or so. With said change, the example code would be:

```
updatePROpts := &bitbucket.UpdatePullRequestOpts{
	Title:    NewStringPointer(c.String("title")),
	Reviewers: []*bitbucket.User{
		{UUID: NewStringPointer("{some_ID}")},
		{UUID: NewStringPointer("{some_ID}")},
	},
}

result, resp, err := api.PullRequests.Update(OWNER, REPO_SLUG, PR_ID, *updatePROpts)
```
You can pass in all the fields of `bitbucket.User` but it seems just the `User.UUID` field is needed and probably better used over `User.Username`.

Though now thinking about it a bit more, perhaps a separate function named `PullRequestsService.AddReviewers` would provide a better experience. But then a `PullRequestsService.RemoveReviewers` might be needed as well to complete the pair. These two proposed functions would provide a handy way to update reviewers separate from the `PullRequestsService.Update` function but still call the same endpoint as there isn't a sub-endpoint under `/2.0/repositories/{username}/{repo_slug}/pullrequests/{pull_request_id}`.

Let me know what you think.

Thanks!

**Comment by davidji99** [comment]
Added the first option as part of #37 to unblock you.

**Comment by dcu** [comment]
thanks a lot

**Comment by dcu** [comment]
`Close source branch` is set to false automatically when using the update PR method

**Comment by davidji99** [comment]
@dcu Ah, that's probably the default value that Bitbucket sets if the JSON payload doesn't have `close_source_branch`.

Looking at that area of code more, I need to consolidate `CreatePullRequestOpts` with `UpdatePullRequestOpts` in a PR so you can set `close_source_branch`.

**Comment by davidji99** [comment]
Going to close this issue for now. If you have any other questions, please open a new issue. Thanks!
